docs/user/sbt.rst

```diff
@@ -168,28 +168,80 @@ package resolution system.
 .. _sonatype: https://github.com/xerial/sbt-sonatype
 .. _bintray: https://github.com/sbt/sbt-bintray
 
-Using 3rd Party libraries with Native Code
-------------------------------------------
+Including Native Code in your Application or Library
+----------------------------------------------------
 
-Third party libraries that are targeted only for the Scala Native platform
-can have C and/or C++ components included in the dependency. The code is
-added to `src/main/resources` and is published like a normal Scala library.
+Scala Native uses native C and C++ code to interact with the underlying
+platform and operating system. Since the tool chain compiles and links
+the Scala Native system, it can also compile and link C and C++ code
+included in an application project or a library that supports Scala
+Native that includes C and/or C++ source code.
 
-If the dependency contains native code, Scala Native will unpack the library,
-compile, link, and optimize any native code along with the Scala Native
-runtime and your application code.
+Supported file extensions for native code are `.c`, `.cpp`, and `.S`.
 
-Including Native Code in your Project
-------------------------------------------
+Note that `.S` files or assembly code is not portable across different CPU
+architectures so conditional compilation would be needed to support
+more than one architecture. You can also include header files with
+the extensions `.h` and `.hpp`.
+
+Applications with Native Code
+-----------------------------
 
 In order to create standalone native projects with native code use the
-following procedure.
+following procedure. You can start with the basic Scala Native template.
+
+Add C/C++ code into `src/main/resources/scala-native`. The code can be put in
+subdirectories as desired inside the `scala-native` directory. As an example,
+create a file named `myapi.c` and put it into your `scala-native` directory
+as described above.
+
+.. code-block:: scala
```

Comment:
Why is the `code-block` language `scala`.  I would have expected `c`.  I think 
Sphinx supports that out-of-the-box.   

Good that you provided the name "myapi.c" just above to help me orient on
language.  Sorry, I get confused easily.

Reply:
Yes, I changed that to C and the `long long` changed to be highlighted in red, like the Scala types. Nice catch.